tedge/src/cli/connect/command.rs

```diff
@@ -176,7 +184,40 @@ where
 // the check can finish in the second try as there is no error response in the first try.
 
 #[tokio::main]
-async fn check_connection_c8y(port: u16) -> Result<(), ConnectError> {
+async fn check_connection_c8y(port: u16) -> Result<DeviceStatus, ConnectError> {
+    for i in 0..2 {
+        print!("Try {} / 2: Sending a message to Cumulocity. ", i + 1,);
+
+        let create_device_result = create_device(port);
+        match create_device_result.await {
+            Ok(DeviceStatus::New) => {
+                return Ok(DeviceStatus::New);
+            }
+            Ok(DeviceStatus::AlreadyExists) => {
+                println!("Received expected response message, connection check is successful.\n",);
+                if i == 0 {
+                    return Ok(DeviceStatus::AlreadyExists);
+                } else {
+                    return Ok(DeviceStatus::New);
```

Comment:
Better put a comment here why are we returning new status here.

Reply:
I think the device status is Unknown here. He might be new but not necessarily: our first message might have been lost.

Reply:
Returning `Unknown` while receiving an `AlreadyExists` response would be confusing. So I renamed the `Unknown` variable as `MightBeNew`.